### PR TITLE
Add step for emitting rate limits for a token

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+parameters:
+  - name: GitHubUser
+    type: string
+  - name: GitHubToken
+    type: string
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml:5]: Should we also add some context of where we are logging from? So we can start adding this to various places to see what the limits look like in different contexts?

File: eng/pipelines/templates/steps/sync-directory.yml [hunk 2]
```diff
@@ -170,3 +170,8 @@ steps:
         PushArgs: -f
         WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
         ScriptDirectory: $(System.DefaultWorkingDirectory)/$(Build.Repository.Name)/${{ parameters.ScriptDirectory }}
+
+- template: /eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml
+  parameters:
+    GitHubUser: azure-sdk
+    GitHubToken: $(azuresdk-github-pat)
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/sync-directory.yml:177]:
```suggestion
    GitHubToken: $(azuresdk-github-pat)

```

**Review by weshaggard — APPROVED** [review]
One suggestion but otherwise looks good.

File: eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml (hunk 1 above)
Comment (benbp) [reply eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml:5]: That may already be included in the metric metadata as we process the build info at the same time

**Review by JimSuplizio — CHANGES_REQUESTED** [review]
@hallipr we have a [git-helpers.ps1](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Helpers/git-helpers.ps1) PowerShell script where we've been adding common git functions and I would rather see this added in there, as common code than inline. You might even be able to reuse or slightly modify some of the existing code to add this particular scenario.

**Comment by hallipr** [comment]
> @hallipr we have a [git-helpers.ps1](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Helpers/git-helpers.ps1) PowerShell script where we've been adding common git functions and I would rather see this added in there, as common code than inline. You might even be able to reuse or slightly modify some of the existing code to add this particular scenario.

I'll open another PR to move the code to the helpers script.

File: eng/pipelines/templates/steps/sync-directory.yml (hunk 2 above)
Comment (hallipr) [reply eng/pipelines/templates/steps/sync-directory.yml:177]: I'll add the line break in another PR when I move the inline script to the github helpers script

File: eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml (hunk 1 above)
Comment (hallipr) [reply eng/common/pipelines/templates/steps/emit-rate-limit-metrics.yml:5]: We'd be applying false granularity to the metric.   It'd be context around the job that's checking the metric while the metric being checked is more broad than that context.

**Pull request merged**